### DISCUSSION TITLE
The universality of metabolic tasks list

### DISCUSSION THREAD

**Discussion by yangjingyu1108** [SysBioChalmers/Human-GEM#733] [Q&A] [answered]
When we use tINIT to generate a model of any tissue or organ, do we need to complete each task in the task list? Can we ignore the metabolic heterogeneity of different cells？

**Comment by mihai-sysbio** [comment]
Hi @yangjingyu1108 , thanks for opening this discussion. To start with, I'm going to link this very related issue https://github.com/SysBioChalmers/Human-GEM/issues/280

**Comment by yangjingyu1108** [comment]
> Hi @yangjingyu1108 , thanks for opening this discussion. To start with, I'm going to link this very related issue #280

Thank you for your answer. I have read the relevant content, but I still don't quite understand. Can you explain it in more detail?

**Comment by JonathanRob** [comment]
Hi @yangjingyu1108, the short answer is no, you don't _have_ to complete any tasks in the list. You can omit the task list entirely and run tINIT without requiring any of the tasks, if you want. This is however generally not recommended because the resulting model will likely not be properly "connected"; i.e., it will not be able to produce biomass, among other functions considered essential to a living cell.

I'm not sure I fully understand what you are asking in your second sentence:

> Can we ignore the metabolic heterogeneity of different cells?

Are you asking if it is permitted by the algorithm, or if it is recommended by the scientific field?

**Comment by yangjingyu1108** [comment]
> Hi @yangjingyu1108, the short answer is no, you don't _have_ to complete any tasks in the list. You can omit the task list entirely and run tINIT without requiring any of the tasks, if you want. This is however generally not recommended because the resulting model will likely not be properly "connected"; i.e., it will not be able to produce biomass, among other functions considered essential to a living cell.
> 
> I'm not sure I fully understand what you are asking in your second sentence:
> 
> > Can we ignore the metabolic heterogeneity of different cells?
> 
> Are you asking if it is permitted by the algorithm, or if it is recommended by the scientific field?

Thank you for your reply. In my opinion, different cells have different metabolic functions, that is, they need to complete different metabolic tasks. So I want to ask if the task list in tINIT is common to different cells.  As you suggested, we need to run the task list. However, for different cells, does each task in the list need to be completed?  Are all the tasks in the list necessary for all cells ?

**Comment by JonathanRob** [comment, marked as answer]
All of the tasks in the [`metabolicTasks_Essential.txt`](https://github.com/SysBioChalmers/Human-GEM/blob/main/data/metabolicTasks/metabolicTasks_Essential.txt) task list should, in theory, be absolutely essential for any human cell, regardless of cell type. So in general I would recommend using this as the task list input for tINIT.

The [`metabolicTasks_Full.txt`](https://github.com/SysBioChalmers/Human-GEM/blob/main/data/metabolicTasks/metabolicTasks_Full.txt) list contains more tasks, some of which may be cell-specific, and therefore are not necessarily required for all cell types. I would therefore _not_ recommend using this task file as the set of required tasks for tINIT.

The [`metabolicTasks_CellfieConsensus.txt`](https://github.com/SysBioChalmers/Human-GEM/blob/main/data/metabolicTasks/metabolicTasks_CellfieConsensus.txt) list was translated from another study into the Human-GEM namespace, and is a great resource. However, I do not think we were able to completely map the metabolites/tasks from the original study into Human-GEM, and I have not myself done any testing with this list, so I would use some caution if this is something you're interested in.

You can also choose to remove one or more tasks from the list you use. For example, if you decide to use the essential task list (metabolicTasks_Essential.txt), you can take a look at the different tasks and decide if any of the tasks seem too specific which you would not like to require that the tINIT model must perform. You can either delete the row(s) corresponding to those unwanted task(s), or comment it out with a hash (`#`) in the first column of those row(s).
